plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,287 @@
+	// Wirte to client
+	// status, err := s.session.InsertRecords(deviceidList, measurementsList, dataTypesList, valuesList, timestampList)
+	err = s.WriteRecordsWithTags(recordsWithTags)
+	if err != nil {
+		s.Log.Errorf("IoTDB Write Error: %s", err.Error())
+	}
+	return err
+}
+
+// Find out data type of the value and return it's id in TSDataType, and convert it if nessary.
+func (s *IoTDB) getDataTypeAndValue(value interface{}) (client.TSDataType, interface{}) {
+	switch v := value.(type) {
+	case int32:
+		return client.INT32, v
+	case int64:
+		return client.INT64, v
+	case uint32:
+		return client.INT64, int64(v)
+	case uint64:
+		if s.ConvertUint64To == "ToInt64" {
+			if v <= uint64(math.MaxInt64) {
+				return client.INT64, int64(v)
+			}
+			return client.INT64, int64(math.MaxInt64)
+		} else if s.ConvertUint64To == "ForceToInt64" {
+			return client.INT64, int64(v)
+		} else if s.ConvertUint64To == "Text" {
+			return client.TEXT, fmt.Sprintf("%d", v)
+		} else {
+			s.Log.Errorf("unknown converstaion configuration of 'convertUint64To': %s", s.ConvertUint64To)
+			return client.UNKNOW, int64(0)
+		}
+	case float64:
+		return client.DOUBLE, v
+	case string:
+		return client.TEXT, v
+	case bool:
+		return client.BOOLEAN, v
+	default:
+		s.Log.Errorf("Unknown datatype: '%T' %v", value, value)
+		return client.UNKNOW, int64(0)
+	}
+}
+
+// convert Timestamp Unit accroding to config
+func (s *IoTDB) ConvertTimestampOfMetric(m telegraf.Metric) (int64, error) {
+	if s.TimeStampUnit == "second" {
+		return m.Time().Unix(), nil
+	} else if s.TimeStampUnit == "millisecond" {
+		return m.Time().UnixMilli(), nil
+	} else if s.TimeStampUnit == "microsecond" {
+		return m.Time().UnixMicro(), nil
+	} else if s.TimeStampUnit == "nanosecond" {
+		return m.Time().UnixNano(), nil
+	}
+	errorMsg := fmt.Sprintf("IoTDB Configuration Error: unknown TimeStampUnit: %s", s.TimeStampUnit)
+	s.Log.Errorf(errorMsg)
+	return 0, errors.New(errorMsg)
+}
+
+// convert Metrics to Records with tags
+func (s *IoTDB) ConvertMetricsToRecordsWithTags(metrics []telegraf.Metric) (*RecordsWithTags, error) {
+	var deviceidList []string
+	var measurementsList [][]string
+	var valuesList [][]interface{}
+	var dataTypesList [][]client.TSDataType
+	var timestampList []int64
+	var tagsList [][]*telegraf.Tag
+
+	for _, metric := range metrics {
+		// write `metric` to the output sink here
+		var tags []*telegraf.Tag
+		tags = append(tags, metric.TagList()...)
+		// deal with basic paramter
+		var keys []string
+		var values []interface{}
+		var dataTypes []client.TSDataType
+		for _, field := range metric.FieldList() {
+			datatype, value := s.getDataTypeAndValue(field.Value)
+			if datatype != client.UNKNOW {
+				keys = append(keys, field.Key)
+				values = append(values, value)
+				dataTypes = append(dataTypes, datatype)
+			}
+		}
+		// Convert timestamp into specified unit
+		ts, tsError := s.ConvertTimestampOfMetric(metric)
+		if tsError != nil {
+			return nil, tsError
+		}
```

Comment:
Please use `err` for errors as this is familiar to readers of the code.

Reply:
ok